### PR TITLE
Completed apis/ GET,PUT,POST,DELETE implementation to a workable status

### PR DESCRIPTION
**Pull request by thilinicooray**
Created utils directory under src/main/java/org.wso2.carbon.apimgt.rest.api to keep util files
Added mappings subfolder under utils to keep dto-to-model mapping utils
Added RestApiUtil.java to keep util functions common for the entire component.

Submitted src contains GET, PUT, POST, DELETE workable implementation for api.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java [hunk 1]
```diff
@@ -49,14 +49,28 @@
...
+        boolean isTenantFlowStarted = false;
+
         try {
-            provider = APIManagerFactory.getInstance().getAPIProvider("admin");
-            apis = provider.searchAPIs(query,type, "admin");
+            String loggedInUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            APIProvider apiProvider = RestApiUtil.getProvider(loggedInUser);
+            String tenantDomain =  CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java]: tenant flow hasn't started yet

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/utils/mappings/APIMappingUtil.java [hunk 2]
```diff
@@ -194,7 +197,7 @@ protected static APIDTO fromAPItoDTO(org.wso2.carbon.apimgt.api.model.API model)
         String transports = StringUtils.join(dto.getTransport(), ',');
         model.setTransports(transports);
         //dto.setType("");   //how to get type?
-        //model.setVisibility(dto.getVisibility().name());
+        model.setVisibility("Public");
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/utils/mappings/APIMappingUtil.java]: It there String constant to replace?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java [hunk 3]
```diff
@@ -65,18 +79,19 @@ public Response apisPost(APIDTO body,String contentType){
...
-             provider.addAPI(apiToAdd);
-
-             //how to add thumbnail
-             //publish to external stores
+            org.wso2.carbon.apimgt.api.model.API apiToAdd = APIMappingUtil.fromDTOtoAPI(body);
+            String loggedInUser = CarbonContext.getThreadLocalCarbonContext().getUsername();
+            APIProvider apiProvider = RestApiUtil.getProvider(loggedInUser);
+            String tenantDomain =  CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java]: tenant flow not started yet

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/utils/mappings/APIMappingUtil.java (hunk 2 above)
Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/utils/mappings/APIMappingUtil.java]: Fixed in #1077 commit aeb16f3

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java (hunk 1 above)
Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java]: The tenants flows have been moved currently from impl level to CXF handler level
Fixed comments in #1091 commit : 897ef52

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java (hunk 3 above)
Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java]: The tenants flows have been moved currently from impl level to CXF handler level
Fixed comments in #1091 commit : 897ef52

File: components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java [hunk 4]
```diff
@@ -38,8 +40,6 @@
 import java.util.ArrayList;
 import java.util.List;
 
-import static org.wso2.carbon.apimgt.rest.api.impl.MappingUtil.fromAPItoDTO;
-import static org.wso2.carbon.apimgt.rest.api.impl.MappingUtil.fromDTOtoAPI;
 
 public class ApisApiServiceImpl extends ApisApiService {
```
Comment (lafernando) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java]: Add class level comments, fix in all the places.

Comment (thilinicooray) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java]: Please note that class comments are available in the up-to-date source in https://github.com/wso2/carbon-apimgt/blob/release-1.10.x/components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java

**Pull request merged**